src/chain/chain.rs

```diff
@@ -438,8 +449,12 @@ impl Chain {
     pub fn remove_elder(&mut self, pub_id: PublicId) -> Result<EldersInfo, RoutingError> {
         self.assert_no_prefix_change("remove elder");
 
-        let mut elders = self.state.new_info.members().clone();
-        let _ = elders.remove(&pub_id);
+        let mut elders = self.state.new_info.p2p_members().clone();
+        let connection_info = self
+            .get_member_connection_info(&pub_id)
+            .ok_or(RoutingError::InvalidStateForOperation)?;
+        let p2p_node = P2pNode::new(pub_id, connection_info);
```

Comment:
Oh nice, I'll go with the section option of implementing `Borrow<PublicId>`

Reply:
As a note, the Borrow seem to be a bit advanced since it put some implicit constraint on the Ord. I'd probably go with either  BTreeMap<PublicId, ConnectionInfo>, or BTreeMap<PublicId, P2pNode>, as that should be sufficient and more explicit that the ordering is only on the PublicId.

Reply:
I take it you mean something like adding 
```
    pub fn member_connection_infos(&self) -> BTreeMap<&PublicId, &ConnectionInfo> {
        self.members
            .iter()
            .map(|p2p_node| (p2p_node.public_id(), p2p_node.connection_info()))
            .collect()
    }
```
to `EldersInfo` and then in `Chain` do
```
        let mut elders = self.state.new_info.member_connection_infos();
        let _ = elders.remove(&pub_id);
        let p2p_elders = elders
            .into_iter()
            .map(|(id, conn_info)| P2pNode::new(*id, conn_info.clone()))
            .collect();
```

Reply:
No, I do not mean add a new map, I mean replace what is a set with a map. (not a map of reference).
But maybe I'm missing something? (could HO to look at code together).

Reply:
You could change the above `member_connection_infos` to return `BTreeMap<PublicId, ConnectionInfo>`, but I'm not sure you'd gain much outside giving the caller no choice about the clone?

For reference the original code called `p2p_members(&self) -> &BTreeSet<P2pNode>`